src/oidcop/oauth2/token.py

```diff
@@ -358,6 +362,284 @@ def post_parse_request(
+                request["subject_token"], grant=True
+            )
+        except (KeyError, UnknownToken):
+            logger.error("Subject token invalid.")
+            return self.error_cls(
+                error="invalid_request",
+                error_description="Subject token invalid"
+            )
+
+        token = _mngr.find_token(_session_info["session_id"], request["subject_token"])
+        if token.is_active() is False:
+            return self.error_cls(
+                error="invalid_request", error_description="Subject token inactive"
+            )
+
+        resp = self._enforce_policy(request, token, config)
+
+        return resp
+
+    def _enforce_policy(self, request, token, config):
+        _context = self.endpoint.server_get("endpoint_context")
+
+        subject_token_types_supported = (
+            self.total_subject_token_types_supported.keys()
+            & config.get("subject_token_types_supported", "urn:ietf:params:oauth:token-type:access_token")
+        )
+        subject_token_types_supported = {
+            k:self.total_subject_token_types_supported[k] for k in subject_token_types_supported
+        }
+
+        if (
+            request["subject_token_type"] in subject_token_types_supported
+            and (
+                subject_token_types_supported[request["subject_token_type"]] != token.token_class
+            )
+        ):
+            return self.error_cls(
+                error="invalid_request", error_description="Wrong token type"
+            )
+
+        if request["subject_token_type"] not in subject_token_types_supported:
+            return TokenErrorResponse(
+                error="invalid_request",
+                error_description="Unsupported subject token type",
+            )
+
+        if (
+            "requested_token_type" in request
+            and request["requested_token_type"] not in config["requested_token_types_supported"]
+        ):
+            return TokenErrorResponse(
+                error="invalid_request",
+                error_description="Unsupported requested token type",
+            )
+
+        request_info = dict(scope=request.get("scope", []))
+        try:
+            check_unknown_scopes_policy(request_info, request["client_id"], _context)
+        except UnAuthorizedClientScope:
+            logger.error("Unauthorized scope requested.")
+            return self.error_cls(
+                error="invalid_grant",
+                error_description="Unauthorized scope requested",
+            )
+
+        try:
+            subject_token_type = request.get("subject_token_type", "")
+            if subject_token_type not in config["policy"]:
+                subject_token_type = ""
+            callable = config["policy"][subject_token_type]["callable"]
+            kwargs = config["policy"][subject_token_type]["kwargs"]
+
+            if isinstance(callable, str):
+                fn = importer(callable)
```

Comment:
wrap this in a separate `try/except` and raise `ImproperlyConfigured` with the appropriate message

Reply:
Only this line (`fn = importer(callable)`) can raise no reason to have the rest in the `try/except` block